dislib/model_selection/_search.py

```diff
@@ -0,0 +1,387 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from collections.abc import Sequence
+from functools import partial
+from itertools import product
+
+from pycompss.api.api import compss_wait_on
+from scipy.stats import rankdata
+from sklearn import clone
+from sklearn.model_selection import ParameterGrid
+import numpy as np
+from sklearn.utils.fixes import MaskedArray
+
+from ._split import infer_cv
+from ._validation import check_scorer, fit_and_score, validate_score, \
+    aggregate_score_dicts
+
+
+class BaseSearchCV(ABC):
+    """Abstract base class for hyper parameter search with cross-validation."""
+
+    def __init__(self, estimator, scoring=None, cv=None, refit=True):
+        self.estimator = estimator
+        self.scoring = scoring
+        self.cv = cv
+        self.refit = refit
+
+    @abstractmethod
+    def _run_search(self, evaluate_candidates):
+        """Abstract method to perform the search. The parameter
+        `evaluate_candidates` is a function that evaluates a ParameterGrid at a
+        time """
+        pass
+
+    def fit(self, x, y=None, **fit_params):
+        """Run fit with all sets of parameters.
+        Parameters
+        ----------
+        x : ds_array
+            Training data samples.
+        y : ds_array, optional (default = None)
```

Comment:
Change to `ds-array` for consistency.

Reply:
Ok. But it breaks PyCharm type hinting. Convention is class names should be CamelCase.